src/main/js/oauth.js

```diff
@@ -292,6 +292,15 @@ gromit.oAuthAuthenticate = function(/*String*/ url, /*function*/ callback, /*boo
     url += '?response_type=token';
     url += '&redirect_uri=' + gromit.getCurrentUrl() + 'oauth.html';
     url += '&client_id=' + gromit.ClientId;
+    
+    if (typeof gromit.AuthParams !== 'undefined') {
+        for (var i = 0; i < gromit.AuthParams.length; i++) {
+            url += '&' + gromit.AuthParams[i].name;
```

Comment:
You probably don't want to add the parameter to the URL if there is a name and no value.

Reply:
A valueless parameter is valid. Rarely used, but may as well allow it.